### DISCUSSION TITLE
Dynamic sets of resources

### DISCUSSION THREAD

**Discussion by frafra** [geopython/pygeoapi#1074] [Ideas]
I develop a plugin which exposes hundreds of resources (same provider but with a different parameter), and a small Python script which updates the configuration file to list all the available resources, which could vary over time. I haven't found a better way to handle that in pygeoapi.

Is this a common need? Like sharing all the files in a folder, or many similar resources coming from the same source.

Would it be an option to read the list of the resources from a database, maybe? Or have groups of dynamic resources?

What do you think?

**Comment by javierurien** [comment]
Hello, would you care to share your implementation? Thanks.

**Reply by frafra** [reply]
Sure, it is nothing fancy: I just generate the YAML when the service starts: https://github.com/NINAnor/nina-pygeoapi/blob/main/src/update_config.py.

The other possibility is to integrate it in Django: https://docs.pygeoapi.io/en/latest/running.html#django. We did that for a more complex project: https://github.com/NINAnor/nina-catalogue/pull/1/files#diff-d617d0f3e36ef6ef6ef319a5ba602b72bb0320d6f5c109d82879eb5d3a399e67.

Relevant discussion: https://github.com/geopython/pygeoapi/issues/1636

**Reply by javierurien** [reply]
Thanks!
